### DISCUSSION TITLE
Thonny newbie- all fine until now a roadblock- "Device is busy or does not respond. Your options:   - wait until ....."

### DISCUSSION THREAD

**Discussion by JBelmont72** [micropython/micropython#13013] [RP2040 / Pico]
Good day.  I'm hitting my head against an obstacle.
I am following the instructions in the  "official Raspberrypi Pico Guide " by Halfacree and Everard.(and cross checking with other resources/tutorials/ reapberrypi.org/github etc)
Working on Botha  Mac air and Mac Pro with updated OS. Load the firmware on the pico fine and have downloaded the app Thonny.  I thought all was great until the middle of running a program with an ADC object( a potentiometer) and PWM controlling led intensity ( and I threw in a bushbutton as well). All was fine and then in the middle of triggering it on with the bushbutton ( had a couple of nested "whiles"). everything got hung up.SInce then I get the same error message over and over. On both computers and happens when I deleted and reloaded Thonny on both computers.  I even used the time machine to go back and start fresh.  
I can run a simple "Hello World " program but as soon as anything that involves a library function, I get this error message.  
Same if I run the program from Thonny or from the Pico.
 "Device is busy or does not respond. Your options:
  - wait until it completes current work;
  - use Ctrl+C to interrupt current work;
  - reset the device and try again;
  - check connection properties;
  - make sure the device has suitable MicroPython / CircuitPython / firmware;
  - make sure the device is not in bootloader mode.
Could not interrupt current process. Please wait, try again or select Stop/Restart!"
Only way to stop I have found is to unplug and start over.  Same results eery time.
I'm enclosing a simple sketch for which it happens- As soon as I press the button in this one, the error message occurs.  Happening with every sketch which worked before but won't now.Enclosing a sketch but happens with all.
import machine
import utime
led_red= machine.Pin(17cc,machine.Pin.OUT)
button = machine.Pin(19,machine.Pin.IN,machine.Pin.PULL_DOWN)
print(button.value())
while True:
    if button.value() == 1:
        led_red.value(True)
        print("Button pressed!")
        print(button.value())
        utime.sleep(1)
    led_red.value(False)       
    print(button.value()

**Comment by GitHubsSilverBullet** [comment]
> * led_red= machine.Pin(**17cc**,machine.Pin.OUT)

›17cc‹ should give you a ›SyntaxError‹ and the last line needs a closing ›)‹
Oh, one more thing: If Thonny doesn't provide you with adequate error messages, use something which does. A simple terminal program will do nicely.

**Comment by bixb922** [comment]
As @GitHubsSilverBullet said, use a terminal emulator like putty, or  check what is happening with mpremote (the official tool), see https://docs.micropython.org/en/latest/reference/mpremote.html

**Comment by shariltumin** [comment]
Don't save your test script as 'main.py', as this will be executed on reboot (soft or hard). If you are at the REPL, you can press Ctrl+C to interrupt a looping program.

Delete the 'main.py' file from the flash. Now save your test program under a different name, for example 'test.py'.

As mentioned by others, use a terminal program to get to the REPL. Now at the REPL prompt '>>>' do ```import test```.


```
>>> import test
...
...
```

You can do the same in Thonny's shell window.

Hope this helps.

**Comment by MrTinkerman** [comment]
You might want to read through [this thread](https://github.com/orgs/micropython/discussions/12686)

As for Thonny, yeah it's a nice convenient GUI to program your devices.
However, it has it's quirks, lots of quirks. You have to be aware of that.

I've been using it for about two years now, pulled out quite some hair and banged my head on my desk a considerable amount of times.
But I'm pretty familiar with it's behavior by now... But I'm not much of a CLI guy, that's why I stuck with it.

Do yourself a favor, don't be like me, focus on learning how to use mpremote or anything else that is not Thonny.

**Comment by JBelmont72** [comment]
These comments have all been wonderful. I am deeply grateful for your help. I’m going to explore all these recommendations. I want know what’s what. Looks like I’ll probably end up going with VS code and using the recommended random nerds tutorial as my entry point. I do have VS code, set up and can do simple programs projects on it. And I do mean simple, thank you

**Reply by MrTinkerman** [reply]
I wish you all the best on your learning journey.

**Comment by JBelmont72** [comment]
I posted my note of gratitude to you all above. Yes.
